servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultHost.java

```diff
@@ -218,59 +218,65 @@ public boolean markExpired() {
+            // Users can apply a ConnectionFactoryFilter if they need to override this "null" value with
+            // TransportObserver.
+            Single<? extends C> establishConnection = connectionFactory.newConnection(address, actualContext, null);
+            if (healthCheckConfig != null) {
+                // Schedule health check before returning
+                establishConnection = establishConnection.beforeOnError(this::markUnhealthy);
+            }
+            return establishConnection
+                    .flatMap(newCnx -> {
+                        if (forceNewConnectionAndReserve && !newCnx.tryReserve()) {
+                            return newCnx.closeAsync().<C>concat(failed(
+                                            Exceptions.StacklessConnectionRejectedException.newInstance(
+                                                    "Newly created connection " + newCnx + " for " + lbDescription
+                                                            + " could not be reserved.",
+                                                    RoundRobinLoadBalancer.class, "selectConnection0(...)")))
+                                    .shareContextOnSubscribe();
+                        }
+
+                        // Invoke the selector before adding the connection to the pool, otherwise, connection can be
+                        // used concurrently and hence a new connection can be rejected by the selector.
+                        if (!selector.test(newCnx)) {
+                            // Failure in selection could be the result of connection factory returning cached
+                            // connection, and not having visibility into max-concurrent-requests, or other threads
+                            // already selected the
+                            // connection which uses all the max concurrent request count.
+
+                            // If there is caching Propagate the exception and rely upon retry strategy.
+                            Single<C> failedSingle = failed(Exceptions.StacklessConnectionRejectedException.newInstance(
                                     "Newly created connection " + newCnx + " for " + lbDescription
-                                            + " could not be reserved.",
-                                    RoundRobinLoadBalancer.class, "selectConnection0(...)")))
-                            .shareContextOnSubscribe();
-                }
+                                            + " was rejected by the selection filter.",
+                                    RoundRobinLoadBalancer.class, "selectConnection0(...)"));
 
-                // Invoke the selector before adding the connection to the pool, otherwise, connection can be
-                // used concurrently and hence a new connection can be rejected by the selector.
-                if (!selector.test(newCnx)) {
```

Comment:
Why this part was removed?
Consider adding a comment in the code clarifying why we do not test `newCnx` now

Reply:
It wasn't - diff mislead?